crates/bifrost/src/providers/replicated_loglet/remote_sequencer.rs

```diff
@@ -140,36 +134,43 @@ where
             payloads,
         };
 
-        let rpc_token = loop {
+        let loglet_commit = loop {
+            let permits = self
+                .record_permits
+                .clone()
```

Comment:
Not sure why this was moved inside the retry loop. Can you share your rationale?